pkg/nginx/listener_test.go

```diff
@@ -0,0 +1,42 @@
+package nginx
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type mockLister struct {
+	interfaces []networkInterface
+	err        error
+}
+
+func (m *mockLister) Interfaces() ([]networkInterface, error) {
+	return m.interfaces, m.err
+}
+
+func TestAllAcceptWgListener(t *testing.T) {
```

Comment:
What about other listener types? `portOnlyListener` / `givenAddressOnlyListener`

Reply:
removed portOnly and wrote test for givenAddressOnlyListener